Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -513,6 +523,23 @@ open class CarPlayNavigationViewController: UIViewController {
         }
     }
     
+    private func checkTunnelState(at location: CLLocation, along progress: RouteProgress) {
+        guard usesDarkStyleWhileInTunnel else { return }
+        let inTunnel = navigationService.isInTunnel(at: location, along: progress)
+        
+        // Entering tunnel
+        if !traversingTunnel, inTunnel {
+            traversingTunnel = true
+            styleManager?.applyStyle(type: .night)
+        }
+        
+        // Exiting tunnel
+        if traversingTunnel, !inTunnel {
+            traversingTunnel = false
+            styleManager?.timeOfDayChanged()
+        }
+    }
+    
```

Comment:
The more features we port to CarPlay, the more duplication we'll have. This method is basically copy of a similar method in `NavigationViewController`. Not sure what's the best way to extract it into different entity, as it uses member variables and there is a chance that we might want to tweak logic for iOS and CarPlay separately.

Reply:
Long term, we should align the two screens so that CarPlayNavigationViewController and NavigationViewController can share a more generic version of NavigationView, probably protocol-based.